File: ClientApp/src/app/core/interfaces/Product/UserInteract/Response/PurchaseDetailOrderResponse.ts
import {WarrantyPreviewResponse} from "@core/interfaces/Product/Response/warrantyPreviewResponse";
import {PurchaseOrderOptionResponse} from "@core/interfaces/Product/UserInteract/Response/PurchaseOrderOptionResponse";

export interface PurchaseDetailOrderResponse {
  id: string;
  productId: string;
  purchaseOptions: PurchaseOrderOptionResponse[];
  orderId: string;
  quoteId: string;
  warrantyId: string;
  warranty: WarrantyPreviewResponse;
}

File: Contracts/Model/Product/UserInteract/Response/PurchaseDetailOrderResponse.cs
using Contracts.Model.Product.Response;

namespace Contracts.Model.Product.UserInteract.Response;

public class PurchaseDetailOrderResponse
{
    public Guid Id { get; set; }
    public Guid ProductId { get; set; }
    public ICollection<PurchaseOrderOptionResponse> PurchaseOptions { get; set; }
    public Guid? OrderId { get; set; }
    public Guid? QuoteId { get; set; }
    public Guid? WarrantyId { get; set; }
    public WarrantyPreviewResponse? Warranty { get; set; }
}